components/backup-stream/src/tempfiles.rs

```diff
@@ -0,0 +1,1006 @@
+// Copyright 2023 TiKV Project Authors. Licensed under Apache-2.0.
+//! This mod provides the ability of managing the temporary files generated by
+//! log backup.
+
+use std::{
+    collections::HashMap,
+    convert::identity,
+    fs::File as SyncOsFile,
+    path::{Path, PathBuf},
+    pin::Pin,
+    sync::{
+        atomic::{AtomicU8, AtomicUsize, Ordering},
+        Arc, Mutex as BlockMutex,
+    },
+    task::{ready, Context, Poll},
+};
+
+use futures::TryFutureExt;
+use kvproto::brpb::CompressionType;
+use tikv_util::warn;
+use tokio::{
+    fs::File as OsFile,
+    io::{AsyncRead, AsyncWrite},
+};
+
+use crate::{
+    annotate,
+    errors::Result,
+    metrics::{
+        IN_DISK_TEMP_FILE_SIZE, TEMP_FILE_COUNT, TEMP_FILE_MEMORY_USAGE, TEMP_FILE_SWAP_OUT_BYTES,
+    },
+    utils::{CompressionWriter, ZstdCompressionWriter},
+};
+
+#[derive(Debug)]
+pub struct Config {
```

Comment:
How about adding to `backup-stream::config::Config`?

Reply:
It seems the deriving the memory quota from temporary file quota is quiet natural. I'm wondering whether exposing this config may bring extra overhead of users' intelligence. Perhaps we can delay adding it before we found someone is needing this in the future?

Reply:
(Anyway we can make it a hidden parameter for now)